pkg/volume/host_path/host_path.go

```diff
@@ -309,3 +323,150 @@ func getVolumeSource(
 
 	return nil, false, fmt.Errorf("Spec does not reference an HostPath volume type")
 }
+
+type hostPathTypeChecker interface {
+	Exists() bool
+	IsFile() bool
+	MakeFile() error
+	IsDir() bool
+	MakeDir() error
+	IsBlock() bool
+	IsChar() bool
+	IsDevice() bool
+	IsSocket() bool
+}
+
+func getFileType(path string) (string, error) {
+	out, err := exec.Command("file", "-b", "--mime-type", path).CombinedOutput()
```

Comment:
oh, don't do this the hard way.  Stat, look at FileInfo.Sys.  Find the docs for what to cast it to and the info you need is in there.

The only wrinkle is you'll have to put this in host_path_linux.go, and a null function in host_path_unsupported.go